test/receiverSpec.js

```diff
@@ -518,6 +519,102 @@ describe('Receiver Module', function () {
+            uuidSeed: Buffer.from('f73e2739c0f0ff4c9b7cac6678c89a59ee6cb8911b39d39afbf2fef9e77bc9c3', 'hex')
+          })
+
+          const psk = this.receiver.generateSharedSecret()
+          const request = sender.createRequest(Object.assign({}, psk, {
+            destination_amount: "1",
+            data: { for: 'that thing' }
+          }))
+
+          this.transfer.account = request.address
+          this.transfer.executionCondition = request.condition
+          this.transfer.data.ilp_header.data.expires_at = request.expires_at
+          this.transfer.data.ilp_header.account = request.address
+          this.transfer.data.ilp_header.data.data = {
+            blob: 'iouIWD_ANIF-NAWUifnawIUfnAIW__aiowna_UWnaw-AWdu'
+          }
+
+          yield expect(this.client.emitAsync('incoming_prepare', this.transfer))
+            .to.be.rejectedWith(/Corrupted ciphertext:/)
+        })
+
+        it('should not fulfill PSK payment with unencrypted memo', function * () {
+          const sender = createSender({
+            client: new MockClient({}),
+            uuidSeed: Buffer.from('f73e2739c0f0ff4c9b7cac6678c89a59ee6cb8911b39d39afbf2fef9e77bc9c3', 'hex')
+          })
+
+          const psk = this.receiver.generateSharedSecret()
+          const request = sender.createRequest(Object.assign({}, psk, {
+            destination_amount: '1',
+            data: { for: 'that thing' }
+          }))
+
+          this.transfer.account = request.address
+          this.transfer.executionCondition = request.condition
+          this.transfer.data.ilp_header.data.expires_at = request.expires_at
+          this.transfer.data.ilp_header.account = request.address
+          this.transfer.data.ilp_header.data.data = {
+            data: { for: 'that thing' }
+          }
+
+          yield expect(this.client.emitAsync('incoming_prepare', this.transfer))
+            .to.be.rejectedWith(/payment request data should only include encrypted blob/)
+        })
+
+        it('should not fulfill payment if reviewPayment callback rejects', function * () {
+          const client = new MockClient({
+            account: 'ilpdemo.blue.bob'
+          })
+
+          const receiver = createReceiver({
+            client: client,
+            hmacKey: Buffer.from('+Xd3hhabpygJD6cen+R/eon+acKWvFLzqp65XieY8W0=', 'base64'),
+            reviewPayment: (paymentRequest) => {
+              return Promise.reject(new Error('rejected!'))
+            }
+          })
+          yield receiver.listen()
+
+          const request = receiver.createRequest({
+            amount: 1,
+            id: '22e315dc-3f99-4f89-9914-1987ceaa906d',
+            expiresAt: this.transfer.data.ilp_header.data.expires_at,
+            data: { for: 'that thing' }
+          })
+
+          this.transfer.executionCondition = request.condition
+          yield expect(client.emitAsync('incoming_prepare', this.transfer))
+            .to.be.rejectedWith('rejected!')
+        })
```

Comment:
There's not any execution path that can trigger that behavior, currently. Either protocol is set to `ipr`, `psk`, or a `"not-my-packet"` rejection is returned.